### ISSUE TITLE
JavaScript heap out of memory running fabric8-ui in dev mode

### ISSUE THREAD

**Issue by jiekang** [openshiftio/openshift.io#2952]
Seen after running fabric8-ui connected to production via
1. `source environments/openshift-prod-cluster.sh`
2. `npm start`

```
<--- Last few GCs --->

[15103:0x41ae010]  6027471 ms: Mark-sweep 1404.4 (1514.6) -> 1404.5 (1503.6) MB, 517.0 / 0.1 ms  (+ 0.0 ms in 0 steps since start of marking, biggest step 0.0 ms, walltime since start of marking 517 ms) last resort GC in old space requested
[15103:0x41ae010]  6027994 ms: Mark-sweep 1404.5 (1503.6) -> 1404.5 (1502.6) MB, 522.2 / 0.1 ms  last resort GC in old space requested


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x3d50482a5ec1 <JSObject>
    2: from [native array.js:1] [bytecode=0x391a1cde72e9 offset=116](this=0x3d50482854f9 <JSFunction Array (sfi = 0x19d7e0727809)>,arrayLike=0x2d9730e26409 <Set map = 0x34cc48d196f9>,mapfn=0x2ad0bec7d2c1 <JSFunction obj.chunks.module.mapChunks.chunk (sfi = 0x2b5d2b426849)>,receiver=0x19d7e0702311 <undefined>)
    3: arguments adaptor frame: 2->3
    4: fnModule(aka fnModule) [/home/jkang/work/gi...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [node]
 2: 0x121837c [node]
 3: v8::Utils::ReportOOMFailure(char const*, bool) [node]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [node]
 5: v8::internal::Factory::NewTransitionArray(int) [node]
 6: v8::internal::TransitionArray::Insert(v8::internal::Handle<v8::internal::Map>, v8::internal::Handle<v8::internal::Name>, v8::internal::Handle<v8::internal::Map>, v8::internal::SimpleTransitionFlag) [node]
 7: v8::internal::Map::CopyReplaceDescriptors(v8::internal::Handle<v8::internal::Map>, v8::internal::Handle<v8::internal::DescriptorArray>, v8::internal::Handle<v8::internal::LayoutDescriptor>, v8::internal::TransitionFlag, v8::internal::MaybeHandle<v8::internal::Name>, char const*, v8::internal::SimpleTransitionFlag) [node]
 8: v8::internal::Map::CopyAddDescriptor(v8::internal::Handle<v8::internal::Map>, v8::internal::Descriptor*, v8::internal::TransitionFlag) [node]
 9: v8::internal::Map::TransitionToDataProperty(v8::internal::Handle<v8::internal::Map>, v8::internal::Handle<v8::internal::Name>, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyAttributes, v8::internal::PropertyConstness, v8::internal::Object::StoreFromKeyed) [node]
10: v8::internal::LookupIterator::PrepareTransitionToDataProperty(v8::internal::Handle<v8::internal::JSObject>, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyAttributes, v8::internal::Object::StoreFromKeyed) [node]
11: v8::internal::Object::AddDataProperty(v8::internal::LookupIterator*, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyAttributes, v8::internal::Object::ShouldThrow, v8::internal::Object::StoreFromKeyed) [node]
12: v8::internal::JSObject::DefineOwnPropertyIgnoreAttributes(v8::internal::LookupIterator*, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyAttributes, v8::internal::Object::ShouldThrow, v8::internal::JSObject::AccessorInfoHandling) [node]
13: v8::internal::Runtime_DefineDataPropertyInLiteral(int, v8::internal::Object**, v8::internal::Isolate*) [node]
14: 0x31192628463d
```

**Comment by corinnekrych** [comment]
@jiekang I've also noticed this one on prod-preview after rebasing my branch this morning.
it might be linked to @christianvogt  changes in here: https://github.com/fabric8-ui/fabric8-ui/pull/2703/files
see this issue https://github.com/webpack/webpack/issues/5089
I think pushing those change sto prod only might help.

**Comment by jiekang** [comment]
@christianvogt Is adding the ModuleConcatenationPlugin required for the fix of https://github.com/openshiftio/openshift.io/issues/2915 ?

Making it only used in production may help, or looking at updating Webpack to a version where this is fixed if it exists would probably be worthwhile.

If it's not required to fix https://github.com/openshiftio/openshift.io/issues/2915 I would prefer to revisit it as a standalone addition. E.g. at least acquire statistics on it's benefits when enabling/disabling it

**Comment by joshuawilson** [comment]
I just merged https://github.com/fabric8-ui/fabric8-ui/pull/2707. Try updating master.

**Comment by christianvogt** [comment]
@jiekang Moved both of them to prod so this should no longer be encountered for dev. We can look to move to webpack 3.11.0 or remove ModuleConcatenationPlugin onto a new issue and disable for now. While trying to fix out issue I ended up trying many different things and this was one of the plugins I had tried and noticed it was missing; it isn't necessary for #2915 . I saw no negative effects and had used it without issue in several past projects.
